test/components/form/submission.spec.js

```diff
@@ -4,60 +4,137 @@ import { mockLogin } from '../../util/session';
-      ]);
-  });
+    it('sends the correct initial requests - Enketo', () => {
+      testData.extendedForms.createPast(1, { xmlFormId: 'a' });
+      return load('/f/enketo-id/new?st=token', {}, { project: false })
+        .testRequests([
+          { url: '/v1/enketo-ids/enketo-id/form?st=token' }
+        ]);
+    });
 
-  it('sends the correct initial requests for draft submission', () => {
-    testData.extendedForms.createPast(1, { xmlFormId: 'a', webformsEnabled: true, draft: true });
-    return load('/f/enketo-id', {}, { project: false })
-      .respondWithData(() => simpleXml)
-      .testRequests([
-        { url: '/v1/enketo-ids/enketo-id/form' },
-        { url: ({ pathname }) => pathname.should.match(/v1\/test\/[a-zA-Z0-9]{64}\/projects\/1\/forms\/a\/draft.xml/) }
-      ]);
+    it('sends the correct initial requests for draft submission', () => {
+      testData.extendedForms.createPast(1, { xmlFormId: 'a', webformsEnabled: true, draft: true });
+      return load('/f/enketo-id', {}, { project: false })
+        .respondWithData(() => simpleXml)
+        .testRequests([
+          { url: '/v1/enketo-ids/enketo-id/form' },
+          { url: ({ pathname }) => pathname.should.match(/v1\/test\/[a-zA-Z0-9]{64}\/projects\/1\/forms\/a\/draft.xml/) }
+        ]);
+    });
   });
 
-  it('renders Enketo Iframe', async () => {
-    testData.extendedForms.createPast(1, { xmlFormId: 'a' });
+  describe('renders the Form', () => {
+    beforeEach(() => {
+      mockLogin();
+      testData.extendedProjects.createPast(1);
+    });
+
+    it('renders Enketo Iframe', async () => {
+      testData.extendedForms.createPast(1, { xmlFormId: 'a' });
+
+      const app = await load('/f/enketo-id', {}, { project: false })
+        .complete();
+
+      const iframe = app.find('iframe');
+
+      iframe.exists().should.be.true;
+    });
 
-    const app = await load('/f/enketo-id', {}, { project: false })
-      .complete();
+    it('renders new Web Form', async () => {
+      testData.extendedForms.createPast(1, { xmlFormId: 'a', webformsEnabled: true });
 
-    const iframe = app.find('iframe');
+      const app = await load('/projects/1/forms/a/submissions/new')
+        .respondWithData(() => simpleXml)
+        .complete();
 
-    iframe.exists().should.be.true;
+      const webForm = app.find('.odk-form');
+
+      webForm.exists().should.be.true;
+    });
   });
 
-  it('renders new Web Form', async () => {
-    testData.extendedForms.createPast(1, { xmlFormId: 'a', webformsEnabled: true });
+  describe('hasAccess', () => {
+    describe('data collector', () => {
+      beforeEach(() => {
+        mockLogin({ role: 'none' });
+        testData.extendedProjects.createPast(1, { role: 'formfill' });
+        testData.extendedForms.createPast(1, { xmlFormId: 'a', webformsEnabled: true });
+      });
+
+      it('can access new submission page', () =>
+        load('/projects/1/forms/a/submissions/new')
+          .respondWithData(() => simpleXml)
```

Comment:
You could get `load()` to return `simpleXml` automatically if that'd help at all with these tests. Even if the XML is only returned in a subset of tests, it's now possible to tell `load()` to return a response conditionally rather than all the time (there are examples of this in test/util/http/data.js).